linkerd/app/outbound/src/discover.rs

```diff
@@ -68,109 +69,109 @@ pub(crate) fn resolver(
        + Send
        + Sync
        + 'static {
-    svc::mk(move |addr: OrigDstAddr| {
+    svc::mk(move |OrigDstAddr(orig_dst)| {
         let profile = profiles
             .clone()
-            .get_profile(profiles::LookupAddr(addr.0.into()));
-        let policy = policies.get_policy(addr.0.into());
+            .get_profile(profiles::LookupAddr(orig_dst.into()));
+        let policy = policies.get_policy(orig_dst.into());
         Box::pin(async move {
             let (profile, policy) = tokio::join!(profile, policy);
+
             let profile = profile.unwrap_or_else(|error| {
                 tracing::warn!(%error, "Error resolving ServiceProfile");
                 None
             });
 
-            let policy_error = match policy {
+            // If there was a policy resolution, return it with the profile so
+            // the stack can determine how to switch on them.
+            match policy {
                 Ok(policy) => return Ok((profile, policy)),
-                Err(error) => error,
-            };
-
-            // Was the policy resolution error a gRPC `NotFound` response?
-            let policy_not_found = policy_error
-                .downcast_ref::<tonic::Status>()
-                .map(|status| status.code() == tonic::Code::NotFound)
-                .unwrap_or(false);
-
-            // If the policy controller responded with `NotFound`, synthesize a
-            // policy, either from the ServiceProfile if one was resolved, or
-            // for a forward to the original destination address.
-            if policy_not_found {
-                if let Some(profile) = profile {
-                    let mut profile_rx = watch::Receiver::from(profile.clone());
-                    let ewma = policy::Load::PeakEwma(policy::PeakEwma {
-                        default_rtt: crate::http::logical::profile::DEFAULT_EWMA.default_rtt,
-                        decay: crate::http::logical::profile::DEFAULT_EWMA.decay,
-                    });
-                    let policy_from_profile = move |profile: &profiles::Profile| {
-                        let dispatcher = if let Some((addr, meta)) = profile.endpoint.clone() {
-                            policy::BackendDispatcher::Forward(addr, meta)
-                        } else if let Some(profiles::LogicalAddr(ref addr)) = profile.addr {
-                            policy::BackendDispatcher::BalanceP2c(
-                                ewma,
-                                policy::EndpointDiscovery::DestinationGet {
-                                    path: addr.to_string(),
-                                },
```

Comment:
This case isn't meaningful because we expect the policy control to always return for anything we could balance over.

Reply:
It turns out this was load-bearing for integration tests: when profiles provided a logical name, we would proceed with balancing/discovery on that name. But this isn't how we expect things to work in practice: We want to rely on the policy controller to provide service information. If it doesn't know about a service, we expect to fallback to a forward stack.